ignite/contrib/handlers/param_scheduler.py

```diff
@@ -54,11 +54,11 @@ def __init__(
         self.optimizer = optimizer
         self.param_group_index = param_group_index
         self.param_name = param_name
-        self.save_history = save_history
         self.event_index = 0
+        self._save_history = save_history
         self._state_attrs = ["event_index", "param_name", "save_history", "param_group_index"]
 
-    def __call__(self, engine: Engine, name: Optional[str] = None):
+    def __call__(self, engine: Optional[Engine], name: Optional[str] = None) -> None:
```

Comment:
I wonder what would be better: either `engine: Optional[Engine] = None` or remove `Optional` and set up a fake engine in `simulate_values`. What do you think ?

Reply:
I think the `Optional` part is totally fine here. When we setup a fake Engine, then the bottom part will just store the parameter-value pairs in the history of the engine state.